### PR TITLE
Fix typo in generics book

### PR DESCRIPTION
**Pull request by omochi**
I found some typos in generics book.

### CODE REVIEW

File: docs/Generics/generics.tex [hunk 1]
```diff
@@ -4888,7 +4888,7 @@ \section{Primary Archetypes}\label{archetypesubst}
 
 \paragraph{Archetypes are not ``inherited''} There's a potential pitfall worth mentioning. Recall that when generic declarations nest, the inner declaration inherits the generic parameters and requirements of the outer declaration, possibly adding new generic parameters or requirements:
 \begin{Verbatim}
-func myAlgorithm<S: Sequence>(_ seq: S) where S.Element: Comparable {
+func complexAlgorithm<S: Sequence>(_ seq: S) where S.Element: Comparable {
```
Comment (omochi) [inline docs/Generics/generics.tex:4891]: This function is referred as `complexAlgorithm` on explanation text below.

Comment (slavapestov) [reply docs/Generics/generics.tex:4891]: This makes the line in the code display too long, so let's change the other reference to myAlgorithm instead.

**Comment by slavapestov** [comment]
The other changes look good, thanks!

**Comment by omochi** [comment]
@slavapestov I updated patch to change `complexAlgorithm` in text to `myAlgorithm` which is same with example code.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**